consensus/src/round_manager.rs

```diff
@@ -587,6 +587,22 @@ impl RoundManager {
             .author()
             .expect("Proposal should be verified having an author");
 
+        let payload_len = proposal.payload().map_or(0, |payload| payload.len());
+        let payload_size = proposal.payload().map_or(0, |payload| payload.size());
+        ensure!(
+            payload_len as u64 <= 2 * self.proposal_generator.max_block_txns(),
```

Comment:
I think this needs to be a separate property. What if someone reduces their block size to 500 locally?

Reply:
well they shouldn't, I think that's an incentive for them to not change the block size

Reply:
@zekun000 - Can you clarify what is the problem we are trying to solve with this check? If someone reduces the block size to be 500, then they will fail this check and not vote on the proposal, but this is not helping anyone right? Since missed vote currently is not directly being counted towards reward performance.